### PR TITLE
Fix symfony/amazon-mailer constraint

### LINKED ISSUE #38000: Can not update symfony/amazon-mailer and symfony/validator to 5.1.4
**Symfony version(s) affected**: 5.1.4

**Description**  
https://packagist.org/packages/symfony/amazon-mailer#v5.1.4
https://packagist.org/packages/symfony/validator#v5.1.4
Amazon-Mailer 5.1.4 needs symfony/mailer:^5.2 which has not yet been released.
Validator 5.1.4 needs symfony/deprecation-contracts: ^2.0 

**How to reproduce**  
composer.json for symfony/amazon-mailer
```
...
 "symfony/amazon-mailer": "^5.1",
 "symfony/mailer": "^5.1",
...
```

composer.json for symfony/validator
```
...
 "symfony/deprecation-contracts": "^2.1",
 "symfony/validator": "^5.1",
...
```

**Possible Solution**  

**Additional context**

### PR DESCRIPTION
**Pull request by michaljusiega**
| Q             | A
| ------------- | ---
| Branch?       | 5.1
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #38000
| License       | MIT
| Doc PR        | 

See #38000 

~~symfony/valdiator left - I need to understand why~~

### CODE REVIEW

**Comment by michaljusiega** [comment]
I don't know why tests failed. :(

File: src/Symfony/Component/Validator/composer.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
         "symfony/polyfill-mbstring": "~1.0",
         "symfony/polyfill-php80": "^1.15",
         "symfony/translation-contracts": "^1.1|^2",
-        "symfony/deprecation-contracts": "^2.0"
+        "symfony/deprecation-contracts": "^2.1"
```
Comment (derrabus) [inline src/Symfony/Component/Validator/composer.json:24]: The first `symfony/deprecation-contracts` was 2.1.0, so your version constraint is a bit more accurate, but I don't expect your change to have any effect.

File: src/Symfony/Component/Mailer/Bridge/Amazon/composer.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "require": {
         "php": ">=7.2.5",
         "symfony/deprecation-contracts": "^2.1",
-        "symfony/mailer": "^5.2"
+        "symfony/mailer": "^4.4|^5.0"
```
Comment (xabbuh) [inline src/Symfony/Component/Mailer/Bridge/Amazon/composer.json:21]: changing the constraint to `^4.4.12|^5.1.4` should help as it only allows to install releases of the Mailer component that contain #37888

**Review by xabbuh — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Validator/composer.json (hunk 1 above)
Comment (michaljusiega) [reply src/Symfony/Component/Validator/composer.json:24]: Apparently yes, but other components have version 2.1 :)

File: src/Symfony/Component/Mailer/Bridge/Amazon/composer.json (hunk 2 above)
Comment (michaljusiega) [reply src/Symfony/Component/Mailer/Bridge/Amazon/composer.json]: Done.

File: src/Symfony/Component/Validator/composer.json (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Component/Validator/composer.json:24]: If it doesn't add anything to solve #38000, I would revert this change.

Comment (michaljusiega) [reply src/Symfony/Component/Validator/composer.json]: @xabbuh Done.

**Review by xabbuh — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @michaljusiega.

**Pull request merged**